### PR TITLE
Set other 'remove' functions to %newobject in Swig.

### PR DESCRIPTION
**Pull request by luciansmith**
Went through and searched for all non-package '::remove' functions that returned non-owning pointers; added them all to libsbml.i

Probably still need to do this for the packages, though hopefully not for the autogenerated ones.

Did not add tests for this; they'd necessarily be valgrind-type tests.

### CODE REVIEW

**Comment by luciansmith** [comment]
This should be ready to merge now, unless someone can think of another class of libsbml functions that return pointers to objects that the caller must delete manually.

File: src/bindings/swig/comp-package.i [hunk 1]
```diff
@@ -36,6 +36,15 @@
 
 #ifdef USE_COMP
 
+%newobject removeSubmodel;
+%newobject removePort;
+%newobject removeReplacedElement;
+%newobject removeModelDefinition;
+%newobject removeGeneProduct;
```
Comment (fbergmann) [inline src/bindings/swig/comp-package.i]: does comp really require a removeGeneProduct?

File: src/sbml/Constraint.cpp [hunk 2]
```diff
@@ -427,7 +427,7 @@ void
 Constraint::replaceSIDWithFunction(const std::string& id, const ASTNode* function)
 {
   if (isSetMath()) {
-    if (mMath->getType() == AST_NAME && mMath->getId() == id) {
+    if (mMath->getType() == AST_NAME && mMath->getName() == id) {
```
Comment (fbergmann) [inline src/sbml/Constraint.cpp]: it seems like here you have teh same fix included, that are also in the pull request for the comp fix. Are there any differences here? which one should we use?

Comment (luciansmith) [reply src/sbml/Constraint.cpp]: Bah, sorry--this fix should indeed be part of the other pull request.  Did it get in both?  In the end, this line needs to be 'getName', however that works out.

File: src/bindings/swig/comp-package.i (hunk 1 above)
Comment (luciansmith) [reply src/bindings/swig/comp-package.i]: Nope!  Good catch.

**Comment by luciansmith** [comment]
OK!  The redundant changes have now been merged away.

**Review by fbergmann — APPROVED** [review]
that works for me .. thanks

**Pull request merged**